src/mito2/src/wal/raw_entry_reader.rs

```diff
@@ -72,14 +47,17 @@ impl<S: LogStore> LogStoreRawEntryReader<S> {
         Self { store }
     }
 
-    fn read_region(&self, ns: RaftEngineNamespace, start_id: EntryId) -> Result<RawEntryStream> {
-        let region_id = ns.region_id;
+    fn read_region(
+        &self,
+        ns: RaftEngineProvider,
+        start_id: EntryId,
+    ) -> Result<EntryStream<'static>> {
+        let region_id = RegionId::from_u64(ns.id);
+        let store = self.store.clone();
+
         let stream = try_stream!({
-            // TODO(weny): refactor the `namespace` method.
-            let namespace = self.store.namespace(region_id.into(), &Default::default());
-            let mut stream = self
-                .store
-                .read(&namespace, start_id)
+            let mut stream = store
+                .read(&Provider::RaftEngine(ns), start_id)
```

Comment:
Why does here hardcode the provider to `RaftEngine`?

Reply:
Good catch! I can simplify it.